sled-hardware/src/underlay.rs

```diff
@@ -44,13 +46,38 @@ pub enum Error {
 
     #[error("Could not enumerate physical links")]
     FindLinks(#[from] FindPhysicalLinkError),
+
+    #[error("Could not set linkprop")]
+    SetLinkprop(#[from] SetLinkpropError),
+
+    #[error("Could not get linkprop")]
+    GetLinkprop(#[from] GetLinkpropError),
 }
 
 /// Convenience function that calls
 /// `ensure_links_have_global_zone_link_local_v6_addresses()` with the links
 /// returned by `find_chelsio_links()`.
 pub fn find_nics() -> Result<Vec<AddrObject>, Error> {
```

Comment:
Installinator won't get this change because it doesn't call `find_nics()` directly; it calls `find_chelsio_links()` and `ensure_links_have_global_zone_link_local_v6_addresses()` directly: https://github.com/oxidecomputer/omicron/blob/051f524a63ae5a000b915f187c99e880536616f7/installinator/src/bootstrap.rs#L30-L40

This is because it wants to grab the first `link` to construct a bootstrap address: https://github.com/oxidecomputer/omicron/blob/051f524a63ae5a000b915f187c99e880536616f7/installinator/src/bootstrap.rs#L61-L65

I think (?) I wrote it that way because it didn't look like I could get from an `AddrObject` (returned by `find_nics()`) back to a `PhysicalLink` (required by `BootstrapInterface::GlobalZone.ip()`), but maybe that was wrong? Given an AddrObject, is `PhysicalLink(addrobj.interface().to_string())` its physical link? If so, I can follow up this PR with a change to installinator to use `find_nics()`.

Reply:
> Given an AddrObject, is PhysicalLink(addrobj.interface().to_string()) its physical link?

I believe this is true, yeah. My read of things is that `PhysicalLink` and the word interface are interchangeable, but that `Link` is different because it implements Drop and therefore has ownership consequences.